lib/iris/fileformats/netcdf.py

```diff
@@ -1593,57 +1725,158 @@ def _get_dim_names(self, cube):
             A :class:`iris.cube.Cube` to be saved to a netCDF file.
 
         Returns:
-            List of dimension names with length equal the number of dimensions
-            in the cube.
+            cube_dimensions, mesh_dimensions
+            * cube_dimensions (list of string):
+                A lists of dimension names for each dimension of the cube
+            * mesh_dimensions (list of string):
+                A list of the mesh dimensions of the attached mesh, if any.
+
+        ..note::
+            One of the mesh dimensions will typically also appear in the cube
+            dimensions.
```

Comment:
As noted in discussion, may be worth noting that : the ordering of the return values is the intended order in the file.
And for consistency with that, this should probably return `(mesh_dimensions, cube_dimensions)` instead of the other way around.

Reply:
That might be better. I certainly had to think about it a couple of times when I first came across this... so perhaps your suggestion would be more natural. Plus it reinforces the intended ordering relationship within the file :+1: